homeassistant/components/template/fan.py

```diff
@@ -363,6 +364,8 @@ async def async_update(self):
             # Validate speed
             if speed in self._speed_list:
                 self._speed = speed
+            elif speed == STATE_UNAVAILABLE:
+                self._speed = STATE_UNAVAILABLE
```

Comment:
Speed, oscillating and direction can't be unavailable. Set them to `None`.

Reply:
done. thanks for your comments and help